feat: add plugin bundles

vhyrro opened the pull request:
This pull request implements the concept of "plugin bundles", i.e. collections of plugins that can be batched together under a single name (as well as a single configuration file).

Currently bundles take on the following form:
```toml
[plugins]
yada = "dev"
yadayada = "dev"
foobar = "dev"

[bundles]
mybundle.items = [
    "yada",
    "yadayada",
    "foobar",
]
```

If a `lua/plugins/mybundle.lua` is supplied, the setup will only be executed for that bundle and individual `yada.lua`, `yadayada.lua` and `foobar.lua` files will be ignored (open to discussion on this one).

TODOs:
- [x] Make `bundles.mybundle = []`  `bundles.mybundle.items = []` instead, as we might want to apply extra metadata to a bundle (e.g. the location of the configuration file or `config = false`).

Open to discussion about any part of the design :)

vhyrro committed:
feat: add plugin bundles

mrcjkb commented on lua/rocks-config/init.lua:
```diff
@@ -94,7 +94,29 @@ function rocks_config.setup(user_configuration)
+    local exclude = {}
+
+    if type(config.plugins and config.plugins.bundles) == "table" then
+        for bundle_name, plugins in pairs(config.plugins.bundles) do
```
question: would it make sense to check if each plugin in a bundle is actually installed (and maybe warn if that's not the case)?

vhyrro replied:
I'm unsure what the performance implications of this would be - maybe doing this in the healthcheck would be a better decision? Polling luarocks for the installed plugin list sounds like not the best idea on startup :sweat_smile:

mrcjkb replied:
By "installed", I think I mean "present in the rocks.toml plugins table"

vhyrro replied:
Oh of course, that I can get behind :D

vhyrro committed:
fix: put bundles under separate namespace, allow `config` to also take a string, make bundles nested tables instead of a list

This commit makes the following changes:
- Bundles now are no longer a single list of plugin names,
  but can have extra metadata attached. The format is now
  `bundles.name.items = []`.
- Bundles can now take a `bundles.name.config` table which
  can only be a string pointing to a lua path of the module to invoke.
- All plugins can now accept `config` as a string, which will execute
  the provided module instead of automatically trying to look up the
  name.

vhyrro committed:
chore: fix makefile

vhyrro committed:
fix: show error when a bundle has an invalid plugin

vhyrro committed:
docs(README): add information about bundles

vhyrro commented:
The CI is tripping but the code looks solid for the time being - tried several edge cases. Might be worth a shot to attach a testing library to this at some point.

mrcjkb commented:
> The CI is tripping but the code looks solid for the time being - tried several edge cases. Might be worth a shot to attach a testing library to this at some point.

Fixed CI on main
